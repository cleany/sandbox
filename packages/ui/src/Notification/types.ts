export { Props } from './';
export type Status = 'none' | 'ok' | 'warning' | 'error';
