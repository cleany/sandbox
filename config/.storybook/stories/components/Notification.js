import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { Notification } from 'ui';
import { 
  withKnobs,
  string,
  color,
  select
 } from '@kadira/storybook-addon-knobs';

const statusOptions = {
  none: 'none',
  ok: 'ok',
  warning: 'warning',
  error: 'error',
};

const stories = storiesOf('Notification', module);
stories.addDecorator(withKnobs);

stories
  .addWithInfo(
    'default notification',
    `
      Properties:

      \`message: {string}\`
      
      A node containing a message to show on the notification.

      \`onClick?: {function()}\`

      A callback to handle closing the notification on click.

      \`status?: {'ok' | 'warning' | 'error' | 'none'}\`

      Status of the notification, which determines color of the background box.

      ~~~js
        <Notification
          onClick={this.handleClose}
          status="warning"
          message="All systems failed."
        />
      ~~~
    `, () =>
        <Notification
          onClick={action('Clicked notification onClick')}
          status={select('status', statusOptions, 'warning')}
          message="All systems failed."
        />,
    {
      inline: true,
      propTables: false,
      source: false,
    }
  );
