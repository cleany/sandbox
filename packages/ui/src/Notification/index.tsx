import * as React from 'react';
import Box from './styles';
import { Paragraph } from '../';
import { Status } from './types';

export interface Props extends React.HTMLProps<typeof Notification> {
  message: JSX.Element | string;
  status?: Status;
}
export default function Notification({
  message,
  status,
  ...rest,
}: Props): JSX.Element {
  const stat = status || 'none';
  return (
    <Box
      status={stat}
      boxSize={{ horizontal: 'medium' }}
      pad="small"
      alignItems="center"
      selectable
      {...rest}
    >
      <Paragraph paragraphSize="large" color="white">{message}</Paragraph>
    </Box>
  );
};
