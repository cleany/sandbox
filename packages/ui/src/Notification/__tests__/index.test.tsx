import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Notification from '../';
import { NotificationPropsRequired, NotificationPropsOther } from './__mocks__/notificationMocks.mock';

describe('<Notification /> default', () => {
  it('should render an Notification element with only the required props', () => {
    const wrapper = shallow(
      <Notification {...NotificationPropsRequired} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render an Notification element with other props', () => {
    const wrapper = shallow(
      <Notification {...NotificationPropsOther} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('The onClick prop should trigger with an onClick event', () => {
  let mockFunction;
  let notification;

  beforeEach(() => {
    mockFunction = jest.fn();
    notification = mount(
      <Notification {...NotificationPropsRequired} onClick={mockFunction} />,
    );
  });

  it('Notification requires onClick prop', () => {
    expect(notification.props().onClick).toBeDefined();
  });

  it('Notification rendered with Box - an HTML div element', () => {
    const renderedNotification = notification.find('div').first();
    expect(renderedNotification).toBeDefined();
  });

  it('Notification click calls mockFunction', () => {
    const renderedNotification = notification.find('div').first();
    renderedNotification.simulate('click');
    expect(mockFunction).toBeCalled();
  });
});
