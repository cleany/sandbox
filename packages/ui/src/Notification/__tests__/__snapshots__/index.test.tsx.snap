// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> default should render an Notification element with only the required props 1`] = `
<Styled(Box)
  alignItems="center"
  boxSize={
    Object {
      "horizontal": "medium",
    }
  }
  pad="small"
  selectable={true}
  status="none"
>
  <styled.p
    color="white"
    paragraphSize="large"
  >
    Mock error message
  </styled.p>
</Styled(Box)>
`;

exports[`<Notification /> default should render an Notification element with other props 1`] = `
<Styled(Box)
  alignItems="center"
  boxSize={
    Object {
      "horizontal": "medium",
    }
  }
  onClick={[Function]}
  pad="small"
  selectable={true}
  status="none"
>
  <styled.p
    color="white"
    paragraphSize="large"
  >
    Mock error message
  </styled.p>
</Styled(Box)>
`;
